Source/NSubstitute.Acceptance.Specs/ReturningResults.cs

```diff
@@ -161,6 +192,43 @@ public void Returns_Null_for_any_args_when_class_returned()
             Assert.That(_something.SomeActionWithParams(123, "something else"), Is.Null);
         }
 
+#if NET45 || NET4
+        [Test]
+        public void Return_a_wrapped_async_result()
+        {
+            _something.CountAsync().Returns(3);
+
+            Assert.That(_something.CountAsync(), Is.TypeOf<Task<int>>());
+            Assert.That(_something.CountAsync().Result, Is.EqualTo(3));
+        }
+
+        [Test]
+        public void Returns_a_task_when_passed_in_and_doesnt_wrap_it()
```

Comment:
What do you test in `Returns_a_task_when_passed_in_and_doesnt_wrap_it`. It tests already existing `Returns` extension:  

`public static ConfiguredCall Returns<T>(this T value, T returnThis, params T[] returnThese)`